firebase-firestore/src/main/java/com/google/firebase/firestore/core/Target.java

```diff
@@ -124,7 +124,8 @@ public boolean hasLimit() {
     if (segment == null) return null;
 
     for (Filter filter : filters) {
-      if (filter.getField().equals(segment.getFieldPath())) {
+      if ((filter instanceof FieldFilter)
+          && (((FieldFilter) filter).getField()).equals(segment.getFieldPath())) {
         FieldFilter fieldFilter = (FieldFilter) filter;
```

Comment:
These three lines are used 4 times in this file. Can we add a helper (`getFieldFilters(Path path)` for example)?

Reply:
Done